qute.ls/com.redhat.qute.ls/src/main/java/com/redhat/qute/services/QuteHover.java

```diff
@@ -354,28 +426,31 @@ private CompletableFuture<Hover> doHoverForPropertyPart(Part part, String projec
 				.thenCompose(resolvedJavaType -> {
 					cancelChecker.checkCanceled();
 					if (resolvedJavaType != null) {
-						ResolvedJavaTypeInfo iterableOfResolvedType = resolvedJavaType.isArray() ? null
-								: resolvedJavaType;
-						Hover hover = doHoverForPropertyPart(part, projectUri, resolvedJavaType, iterableOfResolvedType,
-								hoverRequest);
-						return CompletableFuture.completedFuture(hover);
+						ResolvedJavaTypeInfo iterableOfResolvedType = resolvedJavaType.isArray() ? null : resolvedJavaType;
+						return doHoverForPropertyPart(part, projectUri, resolvedJavaType, iterableOfResolvedType, hoverRequest);
 					}
 					return NO_HOVER;
 				});
 	}
 
-	private Hover doHoverForPropertyPart(Part part, String projectUri, ResolvedJavaTypeInfo resolvedType,
+	private CompletableFuture<Hover> doHoverForPropertyPart(Part part, String projectUri, ResolvedJavaTypeInfo resolvedType,
 			ResolvedJavaTypeInfo iterableOfResolvedType, HoverRequest hoverRequest) {
 		// The Java class type from the previous part had been resolved, resolve the
 		// property
 		JavaMemberInfo member = javaCache.findMember(part, resolvedType, projectUri);
 		if (member == null) {
-			return null;
+			return NO_HOVER;
 		}
 		boolean hasMarkdown = hoverRequest.canSupportMarkupKind(MarkupKind.MARKDOWN);
-		MarkupContent content = DocumentationUtils.getDocumentation(member, iterableOfResolvedType, hasMarkdown);
-		Range range = QutePositionUtility.createRange(part);
-		return new Hover(content, range);
+		CompletableFuture<Void> fetchDocsFuture = CompletableFuture.completedFuture(null);

```

Comment:
If member has no Javadoc, it means that JTD LS javadoc command will be called again for a second hover? If yes we should perhaps add a flag in member which says that Javadoc has been resolved.

Reply:
Okay, it should only fetch the documentation on the first call now.